worker/src/response.rs

```diff
@@ -193,6 +194,19 @@ impl Response {
         self
     }
 
+    /// Sets this response's cors headers from the `Cors` struct.
+    pub fn with_cors(mut self, cors: &Cors) -> Result<Self> {
+        cors.apply_headers(self.headers_mut())?;
+        Ok(self)
+    }
+
+    /// Sets this response's cors headers from the `Cors` struct without checking for the
+    /// `apply_headers` `Result`.
+    pub fn with_cors_unsafe(mut self, cors: &Cors) -> Self {
```

Comment:
I generally dislike functions that encourage you to not handle your errors and I don't see many scenarios where someone wouldn't care about the cors headers being applied. I think it'd be better if we got rid of this method and users could ignore the result themselves with:
```rust
let cors = whatever;
cors.apply_headers(&mut response).ok();
```

Reply:
As above, this was driven by me being stuck with the Result#map approach, fair to remove that too!